Calculator/ViewController.swift

```diff
@@ -22,15 +23,70 @@ class ViewController: UIViewController {
         // Dispose of any resources that can be recreated.
     }
 
-    @IBAction func didgitButtonTapped(_ sender: Any) {
-        let button = sender as! UIButton
-        let buttonText = button.titleLabel!.text!
-        if resultArea.text! == "0" {
-            resultArea.text! = buttonText
+    @IBAction func operationButtonTapped(_ sender: Any) {
+        guard let button = sender as? UIButton else {
+            return
+        }
+        
+        
+        if let input = resultArea?.text, let inputValue = Double(input) {
+            calc.inputValue = inputValue
+            if let buttonOperation = button.titleLabel?.text {
+                switch buttonOperation {
+                case "+":
+                    calc.nextOperation = .plus
+                case "-":
+                    calc.nextOperation = .minus
+                case "x":
+                    calc.nextOperation = .multiply
+                case "/":
+                    calc.nextOperation = .divide
+                default:
+                    ()
+                }
+            }
         }
         else {
-            resultArea.text! += buttonText
+            ()
         }
+        
+        resultArea.text = "0"
```

Comment:
Why 0? It's ok to save same number on screen until user start entering next.

Reply:
Yeah. At this moment if I write "resultArea.text = calc.inputValue" then nextNumber just add to same string variable with inputValue. I'm still in reflection how to save inputValue on screen until nextNumber.